src/clojure/nrepl/middleware/truncate_values_and_outputs.clj

```diff
@@ -0,0 +1,99 @@
+(ns nrepl.middleware.truncate-values-and-outputs
```

Comment:
Add a bit of explanations here about the rationale and high-level logic, plus a note we've added this in 0.5.